godot-core/src/builtin/vectors/vector3.rs

```diff
@@ -291,6 +292,32 @@ impl Vector3 {
         }
     }
 
+    /// Returns the spherical linear interpolation between the vector and `to` by the `weight` amount.
+    ///
+    /// The variable `weight` is representing the amount of interpolation, which is on the range of
+    /// 0.0 to 1.0.
+    ///
+    /// Length is also interpolated in the case that the input vectors have different lengths. If both
+    /// input vectors have zero length or are collinear to each other, the method instead behaves like
+    /// and directly calls [`Vector3::lerp`].
```

Comment:
The "directly calls" is an implementation detail (and makes the sentence more complicated to read), no need to mention that.

Instead, it would be nice if you added such comments in the code itself, e.g. in each `if` branch, to explain why this is equivalent to `lerp`.